huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go

```diff
@@ -0,0 +1,753 @@
+				Description: "The time when the code table was updated.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the code table.",
+			},
+		},
+	}
+}
+
+func resourceArchitectureCodeTableCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		createCodeTableHttpUrl = "v2/{project_id}/design/code-tables"
+		createCodeTableProduct = "dataarts"
+	)
+	createCodeTableClient, err := cfg.NewServiceClient(createCodeTableProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	createCodeTablePath := createCodeTableClient.Endpoint + createCodeTableHttpUrl
+	createCodeTablePath = strings.ReplaceAll(createCodeTablePath, "{project_id}", createCodeTableClient.ProjectID)
+
+	createCodeTableOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	createCodeTableOpt.JSONBody = utils.RemoveNil(buildCreateCodeTableParams(d))
+	createCodeTableResp, err := createCodeTableClient.Request("POST", createCodeTablePath, &createCodeTableOpt)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Architecture code table: %s", err)
+	}
+
+	createCodeTableRespBody, err := utils.FlattenResponse(createCodeTableResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("data.value.id", createCodeTableRespBody)
+	if err != nil || id == nil {
+		return diag.Errorf("error creating DataArts Architecture code table: %s is not found in API response", "id")
+	}
+
+	d.SetId(id.(string))
+
+	yes := hasFieldValues(d)
+	if yes {
+		err = insertCodeTableFieldValues(createCodeTableClient, d)
+		if err != nil {
+			return diag.Errorf("error adding field values to DataArts Architecture code table: %s", err)
+		}
+	}
+
+	return resourceArchitectureCodeTableRead(ctx, d, meta)
+}
+
+func buildCreateCodeTableParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"name_ch":           d.Get("name"),
+		"name_en":           d.Get("code"),
+		"directory_id":      d.Get("directory_id"),
+		"code_table_fields": buildCreateCodeTableFieldParams(d),
+		"description":       utils.ValueIngoreEmpty(d.Get("description")),
+	}
+}
+
+func buildCreateCodeTableFieldParams(d *schema.ResourceData) []map[string]interface{} {
+	rawFields := d.Get("fields").([]interface{})
+	fields := make([]map[string]interface{}, len(rawFields))
+	for i, rawField := range rawFields {
+		fieldMap := rawField.(map[string]interface{})
+		field := make(map[string]interface{})
+		field["ordinal"] = i + 1
+		field["name_ch"] = fieldMap["name"].(string)
+		field["name_en"] = fieldMap["code"].(string)
+		field["data_type"] = fieldMap["type"].(string)
+		field["description"] = utils.ValueIngoreEmpty(fieldMap["description"].(string))
```

Comment:
this is more golang style
```suggestion
		field := map[string]interface{}{
			"ordinal": i + 1,
			"name_ch": fieldMap["name"].(string),
			"name_en": fieldMap["code"].(string),
			"data_type": fieldMap["type"].(string),
			"description": utils.ValueIngoreEmpty(fieldMap["description"].(string)),
		}
```

Reply:
fixed